### PR TITLE
Feature/dhcpv6 fingerbank

### PR DESCRIPTION
**Pull request by julsemaan**
# Description

Adds DHCPv6 fingerprinting to PacketFence through the Fingerbank lib
# Impacts

Fingerbank matching
# Dependencies
#941 (as it contains rework of the Fingerbank access)
# Delete branch after merge

YES
# NEWS file entries
## New Features
- Added DHCPv6 fingerprinting though Fingerbank.

### CODE REVIEW

File: sbin/pfdhcplistener [hunk 1]
```diff
@@ -179,7 +180,22 @@ sub dhcp_detector {
 
 sub process_pkt {
     my ( $user_data, $hdr, $pkt ) = @_;
-    listen_dhcp( $pkt, $user_data );
+    use NetPacket::Ethernet qw(ETH_TYPE_IP);
+    use NetPacket::IP;
+    use NetPacket::IPv6;
+    use NetPacket::UDP;
```
Comment (jrouzierinverse) [inline sbin/pfdhcplistener]: Do not include modules in a function

Comment (julsemaan) [inline sbin/pfdhcplistener]: The description says 'not ready'...

**Comment by julsemaan** [comment]
Ready for review once the #941 is merged

File: conf/pf.conf.defaults [hunk 2]
```diff
@@ -620,6 +625,11 @@ collectd_binary=/usr/sbin/collectd
 # Location of the radsniff3 binary. Only necessary to change if you are not running the pre-packaged version.
 radsniff3_binary=/usr/bin/radsniff3
 # 
+# services.radsniff_binary
+# 
+# Location of the radsniff3 binary. Only necessary to change if you are not running the pre-packaged version.
```
Comment (fdurand) [inline conf/pf.conf.defaults]: radsniff3 => p0f

File: conf/pf.conf.defaults [hunk 3]
```diff
@@ -620,6 +625,11 @@ collectd_binary=/usr/sbin/collectd
...
+p0f_binary=/usr/local/bin/p0f
```
Comment (fdurand) [inline conf/pf.conf.defaults:627]: not in usr/local/pf/... ?

**Comment by dwlfrth** [comment]
I'll review that one once we merge #941 since it seems to have some duplicates between the two.

**Comment by dwlfrth** [comment]
Please rebase to eliminate duplicate code between devel and PR (since #941 has been merged)

**Comment by julsemaan** [comment]
Fixed, can be reviewed

File: db/upgrade-5.3.0-5.4.0.sql [hunk 4]
```diff
@@ -1,6 +1,10 @@
 --
 -- PacketFence SQL schema upgrade from 5.3.0 to 5.4.0
 --
+
+ALTER TABLE node ADD COLUMN `dhcp6_fingerprint` VARCHAR(255) DEFAULT NULL AFTER `dhcp_fingerprint`;
+ALTER TABLE node ADD COLUMN `dhcp6_enterprise` VARCHAR(255) DEFAULT NULL AFTER `dhcp_vendor`;
```
Comment (fdurand) [inline db/upgrade-5.3.0-5.4.0.sql]: Must be in upgrade-5.4.0-5.5.0.sql file

**Comment by julsemaan** [comment]
Fixed, re-review

**Comment by julsemaan** [comment]
@dwlfrth is it good ? or very good to be merged ?

**Pull request merged**